trieste/models/gpflow/sampler.py

```diff
@@ -0,0 +1,348 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module is the home of the sampling functionality required by Trieste's
+GPflow wrappers.
+"""
+
+from __future__ import annotations
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflux.layers.basis_functions.fourier_features import RandomFourierFeaturesCosine
+
+from ...data import Dataset
+from ...types import TensorType
+from ...utils import DEFAULTS
+from ..interfaces import (
+    ProbabilisticModel,
+    ReparametrizationSampler,
+    TrajectoryFunction,
+    TrajectorySampler,
+)
+
+
+class IndependentReparametrizationSampler(ReparametrizationSampler):
+    r"""
+    This sampler employs the *reparameterization trick* to approximate samples from a
+    :class:`ProbabilisticModel`\ 's predictive distribution as
+
+    .. math:: x \mapsto \mu(x) + \epsilon \sigma(x)
+
+    where :math:`\epsilon \sim \mathcal N (0, 1)` is constant for a given sampler, thus ensuring
+    samples form a continuous curve.
+    """
+
+    def __init__(self, sample_size: int, model: ProbabilisticModel):
```

Comment:
if the model can be a generic one and does not need to be a `GPflowPredictor` instance then we can move the whole file to the parent folder, no?

Reply:
These reparam sampler all assume a Gaussian posterior and so only work for GPs. Moreover, they dont even work for DeepGP (Seb wrote his own for gpflux)

Reply:
probabilistic neural nets can give a Gaussian posterior, it should work for them..

Reply:
When needed (in another PR) we can move this up and rename it GaussianBatchRepar......